File: Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> m;
        for(int i=0;i<nums.size();i++)
        {
            if(m.find(nums[i])==m.end())
            {
                m[nums[i]]=1;
            }
            else
            {
                m[nums[i]]++;
            }
            if(m[nums[i]]>nums.size()/2)
            {
                return nums[i];
            }
        }
        return -1;
        
    }
};

File: Golang Solutions/Majority Element.go
func majorityElement(nums []int) int {
    m := make(map[int]int)
    for _,n := range nums{
        _,ok:=m[n]
        if !ok {
            m[n]=1
        }else{
            m[n]++
        }
        if m[n]>len(nums)/2{
            return n
        }
    }
    return -1
}